src/raftstore/store/worker/apply.rs

```diff
@@ -550,51 +592,67 @@ impl ApplyDelegate {
         self.id
     }
 
-    fn handle_raft_committed_entries(
-        &mut self,
-        apply_ctx: &mut ApplyContext,
+    pub fn handle_raft_committed_entries(
+        self,
+        mut apply_ctx: ApplyContext,
         committed_entries: Vec<Entry>,
-    ) {
+    ) -> Box<Future<Item = (Self, ApplyContext), Error = (Self, ApplyContext, Error)> + Send> {
```

Comment:
Seems all future items need `(ApplyDelegate, ApplyContext)`. So I prefer make them as a struct.

Reply:
Not exactly.

Reply:
why not using the structure instead of the tuple. It is weird to see and pass so long tuple.

Reply:
They are different for this function's and inner functions' return types, so i don't think use a struct for each function is a good idea. Using tuple makes it more clearly.